slowmovie.py

```diff
@@ -10,205 +10,349 @@
+    os.mkdir(logdir)
 
-if args.start:
-    print('Start at frame %f' %float(args.start))
-    currentPosition = float(args.start)
+if args.random_frames:
+    print('In random mode')
+else:
+    print ('In play-through mode')
 
-# Ensure this matches your particular screen 
-width = 800 
-height = 480 
+# Pick which video to play
 
-inputVid = viddir + currentVideo
+# First, try the --file CLI argument...
+currentVideo = args.file
 
-# Check how many frames are in the movie 
-frameCount = int(ffmpeg.probe(inputVid)['streams'][0]['nb_frames'])
-print("there are %d frames in this video" %frameCount)
-
-while 1: 
-
-    if args.random:
-        frame = random.randint(0,frameCount)
-    else: 
-        frame = currentPosition
-
-    msTimecode = "%dms"%(frame*41.666666)
-        
-    # Use ffmpeg to extract a frame from the movie, crop it, letterbox it and save it as grab.jpg 
-    generate_frame(inputVid, 'grab.jpg', msTimecode, width, height)
-    
-    # Open grab.jpg in PIL  
-    pil_im = Image.open("grab.jpg")
-    
-    # Dither the image into a 1 bit bitmap (Just zeros and ones)
-    pil_im = pil_im.convert(mode='1',dither=Image.FLOYDSTEINBERG)
+# ...then try a random video, if --random-file was selected...
+if not currentVideo and args.random_file:
+    videos = list(filter(is_vid, os.listdir(viddir)))
+    if videos:
+        currentVideo = random.choice(videos)
 
-    # display the image 
-    epd.display(epd.getbuffer(pil_im))
-    print('Diplaying frame %d of %s' %(frame,currentVideo))
-    
-    currentPosition = currentPosition + increment 
-    if currentPosition >= frameCount:
-        currentPosition = 0
-        log = open(logdir + '%s<progress'%currentVideo, 'w')
-        log.write(str(currentPosition))
-        log.close() 
-    
-        thisVideo = movieList.index(currentVideo)
-        if thisVideo < len(movieList)-1:
-            currentVideo = movieList[thisVideo+1]
+# ...then try the nowPlaying file, which stores the currently-playing video...
+if not currentVideo and os.path.isfile('nowPlaying'):
+    with open('nowPlaying') as file:
+        lastVideo = file.readline().strip()
+        if os.path.isfile(lastVideo):
+            currentVideo = lastVideo
         else:
-            currentVideo = movieList[0]
+            os.remove('nowPlaying')
+
+# ...then just pick the first video in the videos directory...
+if not currentVideo:
+    # Scan through video directory until you find a video file
+    videos = os.listdir(viddir)
+    for file in videos:
+        if is_vid(file):
+            currentVideo = file
+            break
+
+# ...if none of those worked, exit.
+if not currentVideo:
+    print('No videos found in video directory')
+    sys.exit()
+
+with open('nowPlaying', 'w') as file:
+    file.write(currentVideo)
+
+# Keep track of what video we're playing and which ones are available
+if not args.loop:
+    videos = list(filter(is_vid, os.listdir(viddir)))
+    fileIndex = videos.index(currentVideo)
```

Comment:
Could we move this to a function and do this check when needed? The contents of this directory may change over the days/weeks so getting the index at the start may be wrong by the time we need it.

Reply:
Looking at this more moving this to a ```get_next_video(viddir, currentVideo=None)``` type function will also eliminate some dup code lines 212 - 218. Essentially if currentVideo is None then just return the 0 indexed video.

Reply:
Beautiful. Implemented in db61009. I also refactored getting a random video file which appeared twice.